src/Metrics/Registrar.h

```diff
@@ -1,32 +1,37 @@
 #pragma once
 
-#include "Type.h"
+#include "MetricsList.h"
+#include "Sink.h"
+#include <map>
+#include <mutex>
 #include <string>
-#include <unordered_set>
+#include <vector>
 
 namespace Metrics {
 
-enum class LogTo { CARBON, LOG_MSG };
-
-using DestList = std::unordered_set<LogTo>;
-
-class ProcessorInterface;
-
-class Processor;
+class Metric;
 
+/// Register and deregister metrics to be reported via a specified sink
+/// threadsafe
 class Registrar {
 public:
-  Registrar(Registrar &&Registrar);
-
-  bool registerMetric(Metric &NewMetric, DestList Destinations);
-
-  Registrar getNewRegistrar(std::string const &MetricsPrefix) const;
+  Registrar() = default;
```

Comment:
This design of `Registrar` now requires the class to know of both "sinks" and metrics list when previously it did not. I do not think it is an improvement.

Reply:
It previously had to know about the `Processor` instead of metrics list. It doesn't need to know about "sinks", just the enum for sinks, which it also had before. I think I can resolve that.